Hazel/Dtls/DtlsConnectionListener.cs

```diff
@@ -279,8 +279,13 @@ private void ProcessIncomingMessage(ByteSpan message, IPEndPoint peerAddress)
                 return;
             }
 
+            List<MessageReader> applicationMessages = new List<MessageReader>();
```

Comment:
I don't think putting this here is a great idea. Allocating a new list for every packet is pretty frequent. Perhaps we make it a ConcurrentBag and tack it onto the PeerData?

Reply:
Yeah, you're right, we should avoid the allocation. The ConcurrentBag strategy looks great to me---kind of interesting that when you add to the bag, a different thread might pull the messages you add to it

I think if/when we move to a lockless scenario we probably won't need this juggle, so noting our fix may be temporary